packages/http/src/mocker/__tests__/HttpMocker.spec.ts

```diff
@@ -676,5 +689,142 @@ describe('mocker', () => {
         });
       });
     });
+
+    describe('should return 500 error', () => {
+      it('returns a 500 error response', () => {
+        jest.spyOn(helpers, 'negotiateOptionsForInvalidRequest').mockReturnValue(
+          right({
+            code: '500',
+            mediaType: 'application/json',
+            bodyExample: mockResource.responses[3].contents![0].examples![0],
+            headers: [],
+          })
+        );
+        const mockResult = mock({
+          config: { dynamic: false },
+          resource: mockResource,
+          input: Object.assign({}, mockInput, {
+            validations: [{ severity: DiagnosticSeverity.Error }],
+          }),
+        })(logger);
+        assertRight(mockResult, result => {
+          expect(result).toMatchObject({
+            statusCode: 500,
+            body: { message: 'Internal Server Error' },
+          });
+        });
+      });
+    });
+
+    describe('With __bundled__ should get response', () => {
```

Comment:
This test is passing with and without your changes so it still doesn't seem to be testing the initial bug you were seeing / the fix hasn't changed anything. It should be removed or changed to properly capture the issue you are seeing and fixing.

We have [test harness tests](https://github.com/stoplightio/prism/blob/master/test-harness/specs/references/E2E-Recursive-Object-Handling.oas3.txt) for circular references and those have been passing in the pipeline so it seems like circular references are getting resolved in most cases. Maybe a test harness test for the specific scenario that is causing you not to see circular references get resolved would help demonstrate the issue a bit better.